tenant/duplicate_reads.go

```diff
@@ -0,0 +1,362 @@
+package tenant
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/kv"
+	"go.uber.org/zap"
+)
+
+var _ influxdb.BucketService = (*dupReadBucketService)(nil)
+var _ influxdb.OrganizationService = (*dupReadOrganizationService)(nil)
+var _ influxdb.UserResourceMappingService = (*dupReadUserResourceMappingService)(nil)
+var _ influxdb.UserService = (*dupReadUserService)(nil)
+
+// readOnlyStore is a wrapper for kv.Store that ensures that updates are not applied.
+type readOnlyStore struct {
```

Comment:
We will very quickly be rolling in a feature flagging system that allows us to write to the "new" tenant service. I suspect the read only store will cause some headaches when we do this.

Reply:
Planning to made this modular, wait for it.

I wouldn't feel safe without it.

I don't want a `Find*` to accidentally issue an `Update`

Reply:
I think its ok to leave in and then we can swap it out when we are ready to write with the new.